vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/cast/AbstractCastToOperation.java

```diff
@@ -226,4 +226,8 @@ public void convertNormalizedString(UTF8StringPointable stringp, DataOutput dOut
     public void convertToken(UTF8StringPointable stringp, DataOutput dOut) throws SystemException, IOException {
         throw new SystemException(ErrorCode.XPTY0004);
     }
+
+    public void convertNull(DataOutput dOut) throws SystemException, IOException {
```

Comment:
Here is an example query: 

xs:string("test") castable as js:null

Does this help to explain? Add a test query with the above syntax and update CastableAsStringOperation similar to CastAsStringOperation.